### PR TITLE
Upgrade to latest JUnit Foundation release

### PR DESCRIPTION
**Pull request by sbabcoc**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,9 +5,10 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.epam.reportportal</groupId>
     <artifactId>agent-java-junit</artifactId>
-    <version>4.0.0</version>
+    <version>4.0.2-SNAPSHOT</version>
```
Comment (avarabyeu) [inline pom.xml]: i would make it 4.1 since we drop JDK7 support

Comment (sbabcoc) [reply pom.xml]: The support for JDK 7 was dropped at version 4.0.0, but other changes in this PR warrant a minor-version bump.

**Comment by sbabcoc** [comment]
Augh! I tweaked the test code to actually perform some sort of operation on the input, and communication with the RP back-end is now randomly failing.

**Comment by sbabcoc** [comment]
I've now made sure that I handle specific invocations of methods properly by pairing them with their respective runners. This is the thing I was missing last time. Since applying these revisions, all of my launch results are captured flawlessly.

**Comment by sbabcoc** [comment]
I finally managed to implement successful operation with j unit 4 running parallel methods. Give this a try and let me know how it works for you.

**Comment by sbabcoc** [comment]
I applied more thread-related fixes to `JUnit Foundation`. This resolved the latest issues Andrei found.

**Pull request merged**